course7/homeworks/patrickkostjens/ls/ls.cc

```diff
@@ -0,0 +1,13 @@
+#include <iostream>
+#include <boost/filesystem.hpp>
+
+int main(int argc, char** argv)
+{
+	boost::filesystem::path current_directory(argc >= 2 ? argv[1] : ".");
+
+	boost::filesystem::directory_iterator end;
+	for (boost::filesystem::directory_iterator itr(current_directory); itr != end; ++itr) 
```

Comment:
This is too old school. Is this because you are using very old version of Boost?
Please, have a look at Waleed's implementation to see how to use range-based-for for this.